sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -660,36 +778,74 @@ case class HashAggregateExec(
       }
     }
 
+
     val updateRowInVectorizedHashMap: Option[String] = {
```

Comment:
This and `updateRowInRowBasedHashMap` seem to have a lot of duplicated code. Can this be avoided?